src/Microsoft.NetCore.Analyzers/Core/Security/ReviewCodeForSqlInjectionVulnerabilities.cs

```diff
@@ -0,0 +1,133 @@
+                                    if (sourceInfoSymbolMap.IsSourceMethod(invocationOperation.TargetMethod))
+                                    {
+                                        requiresTaintedDataAnalysis = true;
+                                    }
+                                },
+                                OperationKind.Invocation);
+
+                            operationBlockStartContext.RegisterOperationBlockEndAction(
+                                operationBlockAnalysisContext =>
+                                {
+                                    if (!requiresTaintedDataAnalysis)
+                                    {
+                                        return;
+                                    }
+
+                                    TaintedDataAnalysisResult taintedDataAnalysisResult = TaintedDataAnalysis.GetOrComputeResult(
+                                        operationBlockAnalysisContext.OperationBlocks[0].GetEnclosingControlFlowGraph(),
+                                        operationBlockAnalysisContext.Compilation,
+                                        operationBlockAnalysisContext.OwningSymbol,
+                                        sourceInfoSymbolMap,
+                                        new TaintedDataSymbolMap<SanitizerInfo>(
+                                            wellKnownTypeProvider,
+                                            PrimitiveTypeConverterSanitizers.SanitizerInfos),
+                                        sinkInfoSymbolMap);
+                                    foreach (TaintedDataSourceSink sourceSink in taintedDataAnalysisResult.TaintedDataSourceSinks)
+                                    {
+                                        if (sourceSink.SinkKind != SinkKind.Sql)
+                                        {
+                                            continue;
+                                        }
+
+                                        foreach (SymbolAccess sourceOrigin in sourceSink.SourceOrigins)
+                                        {
+                                            Diagnostic diagnostic = Diagnostic.Create(
+                                                Rule,
+                                                sourceSink.Sink.Location,
+                                                new Location[] { sourceOrigin.Location },
+                                                sourceSink.Sink.Symbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                                                sourceSink.Sink.AccessingMethod.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                                                sourceOrigin.Symbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                                                sourceOrigin.AccessingMethod.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat));
```

Comment:
Consider adding a comment with the diagnostic's message format, so it is easy for reader to map each argument to the message.

` // CA3001: Potential SQL injection vulnerability was found where '{0}' in method '{1}' may be tainted by user-controlled data from '{2}' in method '{3}'.